airflow/jobs.py

```diff
@@ -824,25 +827,40 @@ def _change_state_for_tis_without_dagrun(self,
         """:type: list[TaskInstance]"""
 
         for task_instance in task_instances_to_change:
-            dag_run = DagRun.get_run(session,
-                                     task_instance.dag_id,
-                                     task_instance.execution_date)
-            if dag_run is None:
+            dag_runs = DagRun.find(dag_id=task_instance.dag_id,
+                                   execution_date=task_instance.execution_date,
+                                   )
+
+            if len(dag_runs) == 0:
                 self.logger.warn("DagRun for %s %s does not exist",
                                  task_instance.dag_id,
                                  task_instance.execution_date)
-            elif dag_run.state != State.RUNNING:
-                self.logger.warn("Setting %s to state=%s as the state of "
-                                 "%s is not %s",
+                continue
+
+            # There should only be one DAG run. Add some logging info if this
+            # is not the case for later debugging.
+            if len(dag_runs) > 1:
+                self.logger.warn("Multiple DagRuns found for {} {}: {}"
+                                 .format(task_instance.dag_id,
+                                         task_instance.execution_date,
+                                         dag_runs))
+
+            dag_is_running = True
+            for dag_run in dag_runs:
+                if dag_run.state == State.RUNNING:
+                    dag_is_running = True
+                    break
+
+            if not dag_is_running:
+                self.logger.warn("Setting %s to state=%s as it does not have "
```

Comment:
Will change, but I noticed there's an inconsistency in our linter. When using `logging.info`, it says to use `%s` instead?